implementations/rust/ockam/ockam_api/src/nodes/models/credentials.rs

```diff
@@ -0,0 +1,70 @@
+//! Credentials request/response types
+
+use minicbor::{Decode, Encode};
+use ockam_core::compat::borrow::Cow;
+use ockam_core::CowStr;
+
+#[cfg(feature = "tag")]
+use ockam_core::TypeTag;
+use ockam_multiaddr::MultiAddr;
+
+#[derive(Clone, Debug, Decode, Encode)]
+#[rustfmt::skip]
+#[cbor(map)]
+pub struct SetAuthorityRequest {
+    #[cfg(feature = "tag")]
+    #[n(0)] tag: TypeTag<1864428>,
+    #[b(1)] pub authorities: Vec<Vec<u8>>,
```

Comment:
Would be nice to have a bit more type information here.

Reply:
What do you mean by "type information"?

Reply:
I mean a more specific type, e.g. a `Vec<PublicIdentity>`.

Reply:
Enabling cbor for `PublicIdentity` would take a lot of time considering how big and composite this struct is + how time consuming is adding cbor attributes

Reply:
Not sure. The `Encode` and `Decode` impls could use the existing `PublicIdentity::export` and `PublicIdentity::import` functionality I guess. It can be done in another PR.

Reply:
1. Then it will use serde bare under the hood
2. `Identity` and `PublicIdentity` types also provide some guarantees about the data they contain, you shouldn't be able to construct those types with invalid inner `ChangeHistory`, this is why their import functions are doing checks, which are not possible to do during simple Decode

Reply:
> this is why their import functions are doing checks, which are not possible to do during simple Decode

Could you not create a `Vault` in a `Decode` impl and use it to do the validation?